keps/sig-node/20190925-enable-seccomp-by-default.md

```diff
@@ -0,0 +1,335 @@
+| `docker/default` 	| The Docker default seccomp profile is used. Deprecated as of Kubernetes 1.11. Use  `runtime/default` instead. 	| Unchanged, Deprecated 	| No 	| N/A 	|
+| `unconfined` 	| Seccomp is not applied to the container processes (the current default in Kubernetes), if no alternative is provided. Usage should be disencouraged. 	| Unchanged 	| No 	| N/A 	|
+
+
+### 3. New Kind for custom profiles
+
+Promote seccomp profiles to first-class citizen as a new *cluster-level* resource, represented by a new `SeccompProfile` API resource. 
+The new API resource would remove the existing overhead of creating and synchronising files across all new and existing nodes. Allowing users to maintain custom profiles by interacting with the API server.
+
+More details on types needed and usage on the [Implementation Details](#implementation-details) section.
+
+ 
+### 4. Set `runtime/default-audit` as the default profile
+
+By default clusters will have `runtime/default-audit` applied to all containers (users can opt-out). This profile is similar to the existing `runtime/default`, with the difference being that syscalls used that are outside the allowed list will be logged instead of having their execution blocked. This improves the default auditing capabilities of high risk syscalls whilst not breaking existing workloads.
+
+An extra benefit is to provide an easier route for users to move to `runtime/default`. They can determine their workload readiness by going through violations on syslogs, instead of having to profile each application individually. 
+
+To support audit mode, some validation will be required ensuring the downstream dependencies (Kernel, Libseccomp and runc) also supports it. If audit mode is not supported, the default seccomp profile will be set back to `unconfined`, for backwards compatibility. 
+
+### User Stories
+
+#### Story 1
+As a user and administrator, I want to be able to assess what pods within my cluster would be affected by applying more restrictive seccomp profiles.
+
+#### Story 2
+As a user, I want to be able to assess whether my pods will be affected by a more restrictive seccomp profiles.
+
+#### Story 3
+As a user and administrator, I want to be able to create and maintain custom seccomp profiles without having to physically copy files across all my existing (and new) nodes.
+
+#### Story 4
+As a user and administrator, I want to be able Kubernetes to audit all potentially dangerous system calls from my containers by default.
+
+
+### Implementation Details
+
+### 1. Audit mode support (Details)
+
+Kubernetes will continue to be unaware of downstream support. If a user tries to use unsupported actions (i.e. `SCMP_ACT_LOG`) today, the lower level dependencies will return an error. As a result, the container won't be able to start. We are proposing no changes to this behaviour.
```

Comment:
There is a caveat here, if the application tries to use an unsupported action it may be able to handle the error. Now if the unsupported action is defined in the seccomp profile, runC will throw an error during the validation process, before starting the container. Here's where they do it: 
https://github.com/blacktop/runc/blob/18e700b4e3f921391472e3e32634e13985c186df/libcontainer/seccomp/seccomp_linux.go#L119

I will amend that extract to make it clearer.